File: 240922/중복되지 않는 숫자 중 최대/max-of-unique-number.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    // 여기에 코드를 작성해주세요.
    int n;
    cin >> n;
    int arr[n] = {};
    int blank[1000] = {};

    for (int i = 0; i < n; i ++) {
        cin >> arr[i];
        blank[arr[i]] += 1;
    }
    vector<int> vec;

    for (int j = 0; j < 1000; j ++) {
        if (blank[j] == 1) {
            vec.push_back(j);
        }
    }

    if (!vec.empty()) {  // check 벡터가 비어 있지 않다면
        cout << vec.back();  // 벡터의 마지막 요소 출력
    } else {
        cout << -1;  // 벡터가 비어 있다면 -1 출력
    }
    

    return 0;
}

File: 240922/중복되지 않는 숫자 중 최대/max-of-unique-number.py
n = int(input())

arr = list(map(int, input().split()))
maxnum = -1
blank = [0] * 1000

for i in range(n):
    if maxnum <= arr[i]:
        blank[arr[i]] += 1
check = []
for j in range(1000):
    if blank[j] == 1:
        check.append(j)

if len(check) != 0:
    print(check[-1])
else :
    print(-1)